packages/sdk/src/mb-impl/token.ts

```diff
@@ -0,0 +1,214 @@
+// Mintbase token contract JS implementation
+
+import { TransactionArgs } from '../calls';
+import {
+  DEFAULT_MB_LOGO,
+  METHOD_NAMES,
+  MB_MAINNET_TOKEN_FACTORY,
+  MB_TESTNET_TOKEN_FACTORY,
+  Network,
+} from './constants';
+import {
+  TransferArgs,
+  BurnArgs,
+  SingleTokenArgs,
+  DeployTokenContractArgs,
+  AccountId,
+  TransferTokenContractOwnership,
+  MintArgs,
+  AddRemoveMinterArgs,
+  BatchChangeMinters,
+  RevokeAccountArgs,
+} from './token.types';
+
+// TODO: how to deal with this differently?
+const getFactoryContract = (network: string): AccountId => {
+  if (network === Network.MAINNET) return MB_MAINNET_TOKEN_FACTORY;
+  else return MB_TESTNET_TOKEN_FACTORY;
+};
+
+// TODO: figure out a way to generate gas and deposit for each
+
+export const transfer = (args: TransferArgs): TransactionArgs => {
+  const { nftContractId, transfer } = args;
```

Comment:
If the agrs to `transfer` have a `transfer` field, that a) feels odd to me, and b) seems to be a very different API design than the `mint` function